plugins/modules/zabbix_host.py

```diff
@@ -376,7 +366,17 @@
   set_fact:
     ansible_zabbix_auth_key: 8ec0d52432c15c91fcafe9888500cf9a607f44091ab554dbee860f6b44fac895
 
-- name: Create a new host or update an existing host's info
+- name: Create a new host or rewrite an existing host's info
+# Set task level following variables for Zabbix Server host in task
+  vars:
+    ansible_network_os: community.zabbix.zabbix
+    ansible_connection: httpapi
+    ansible_httpapi_port: 443
+    ansible_httpapi_use_ssl: true
+    ansible_httpapi_validate_certs: false
+    ansible_zabbix_url_path: 'zabbixeu'  # If Zabbix WebUI runs on non-default (zabbix) path ,e.g. http://<FQDN>/zabbixeu
+  become: false
+  delegate_to: zabbix-example-fqdn.org
```

Comment:
You can use it differently by specifying a `ansible_host` variable. So `ansible_host: zabbix-example-fqdn.org` in the vars property.

Reply:
@dj-wasabi 
ok. try to do it.
usual inventory
hostname ansible_host=123.123.212.3

so if i use ansible_host for zabbix, it will breake all

Reply:
I have this configured:
```
    ansible_user: "{{ zabbix_server_username }}"
    ansible_host: "{{ zabbix_server_hostname }}"
    ansible_httpapi_pass: "{{ zabbix_server_password }}"
    ansible_httpapi_use_ssl: true
    ansible_httpapi_validate_certs: false
    ansible_httpapi_port: 443
```
And all my Zabbix related tasks will work fine and have access to my Zabbix FQDN.

Reply:
`delegate_to:` and `ansible_host:` achieve essentially the same result (or maybe I don't know Ansible internals well enough).
Let's agree on something guys, we need to release new version with Zabbix 6.4 adoption and I'd love documentation "fixes" to go into new version.
@D3DeFi you did not say anything in this discussion -)

Reply:
I would argue that `ansible_host` should only be changed at the play level and not the task.  If I am in the process of adding a new system to the server, then the host I am working on is the client computer, but obviously, we have to modify the server a little bit to make that happen.  When we do that modification we're doing it for the benefit of the client, not the server so the client would be my ansible_host and I would delegate to the server.  Technically, I think either option would work, but I think that leaving the host alone for the duration of the play is the cleaner option.

Reply:
The `ansible_host` is quite important for Ansible (Actually, the most important property) as this tells Ansible on which host the tasks should be executed. In my case I added it as a var to the task, as my playbook is executed on `hosta`. If you set it at host level, then all the tasks are executed on the host that is set in `ansible_host`. So it is very crusial to set the `ansible_host` on the task. And with setting `ansible_host` you can provide the fqdn (especially if FQDN is totally different than the hostname of the instances that Zabbix is running). Ansible runs in my case on ZAB1, ZAB2, but due to Nginx config only available on FQDN.

With the above, the `delegate_to` won't help me, as when a task is delegated the playbook needs 'ssh' access to that host. But, the `delegate_to` is maybe the safest option because of the above message.

My 2 cents. :)

Reply:
@dj-wasabi 
_delegate_to_ does not ssh in this case due to connection method **httpapi**
and task level _ansible_host_ is ok too. but we should not change it on higher level, like play, role or global